File: rename.py
import os
import re

root_dir = 'c:\\Users\\Tejas\\Desktop\\Source-Codes\\MinecraftSTUFF\\vanilla-core'

ignore_dirs = {'.git', '.idea', 'target', 'build', 'node_modules', 'paper-test-server', 'other-plugins', '.docusaurus'}

def replace_in_file(filepath):
    try:
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()
    except Exception:
        return

    orig_content = content

    # Specific URL overrides
    content = content.replace('https://modrinth.com/plugin/vanillacorewastaken', 'https://modrinth.com/plugin/vanillacorewastaken')
    content = content.replace('https://discord.gg/7fQPG4Grwt?utm_source=vanillacore', 'https://discord.gg/7fQPG4Grwt?utm_source=vanillacore')
    
    # Specific SLUG variables
    content = content.replace(\"'vanillacorewastaken'\", \"'vanillacorewastaken'\")
    content = content.replace('\"smpwatchdog\"', '\"vanillacorewastaken\"')

    # General replaces
    content = content.replace('Vanilla Core', 'Vanilla Core')
    content = content.replace('VanillaCore', 'VanillaCore')

    # Update Modrinth ID in configs if it exists
    content = content.replace('GH4H8ndx', 'GH4H8ndx') # If Modrinth ID stays same? wait user said vanillacorewastaken is slug, but ID might change. Let's keep ID same unless told otherwise.

    if orig_content != content:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(content)
        print(f'Updated {filepath}')

for root, dirs, files in os.walk(root_dir):
    dirs[:] = [d for d in dirs if d not in ignore_dirs]
    for file in files:
        if file.endswith(('.jar', '.png', '.gif', '.zip', '.tar.gz', '.class', '.ico', '.woff', '.woff2')):
            continue
        replace_in_file(os.path.join(root, file))

print('Done!')

File: rename.js
const fs = require('fs');
const path = require('path');

const rootDir = 'c:\\Users\\Tejas\\Desktop\\Source-Codes\\MinecraftSTUFF\\vanilla-core';
const ignoreDirs = new Set(['.git', '.idea', 'target', 'build', 'node_modules', 'paper-test-server', 'other-plugins', '.docusaurus']);

function walk(dir) {
    let results = [];
    const list = fs.readdirSync(dir);
    list.forEach(function(file) {
        if (ignoreDirs.has(file)) return;
        const filePath = path.join(dir, file);
        const stat = fs.statSync(filePath);
        if (stat && stat.isDirectory()) {
            results = results.concat(walk(filePath));
        } else {
            if (!file.endsWith('.jar') && !file.endsWith('.png') && !file.endsWith('.gif') && !file.endsWith('.zip') && !file.endsWith('.tar.gz') && !file.endsWith('.class') && !file.endsWith('.ico')) {
                results.push(filePath);
            }
        }
    });
    return results;
}

const files = walk(rootDir);

files.forEach(filePath => {
    try {
        const origContent = fs.readFileSync(filePath, 'utf8');
        let content = origContent;
        
        content = content.replace(/https:\/\/modrinth\.com\/plugin\/smpwatchdog/g, 'https://modrinth.com/plugin/vanillacorewastaken');
        content = content.replace(/https:\/\/discord\.gg\/7fQPG4Grwt\?utm_source=smpwatchdog/g, 'https://discord.gg/7fQPG4Grwt?utm_source=vanillacore');
        content = content.replace(/'vanillacorewastaken'/g, "'vanillacorewastaken'");
        content = content.replace(/"vanillacorewastaken"/g, '"vanillacorewastaken"');
        content = content.replace(/Vanilla Core/g, 'Vanilla Core');
        content = content.replace(/VanillaCore/g, 'VanillaCore');

        if (content !== origContent) {
            fs.writeFileSync(filePath, content, 'utf8');
            console.log('Updated ' + filePath);
        }
    } catch (e) {}
});
console.log('Done!');
